python/piernik_setup.py

```diff
@@ -332,11 +335,10 @@ def setup_piernik(data=None):
     cmd = "echo '#include \"%spiernik.h\"' > \"%s\"" % ('src/base/', foo_path)
     cmd += " && cpp %s -dM -I%s \"%s\" && rm \"%s\"" % (
         cppflags, probdir, foo_path, foo_path)
-    try:
+    if sys.version_info >= (3,0,0):
         defines = sp.Popen([cmd], stdout=sp.PIPE,
```

Comment:
You can use `get_stdout` here too.

`defines = get_stdout(cmd).rstrip().split("\n")`